lib/src/file_util.rs

```diff
@@ -146,6 +162,51 @@ pub fn persist_content_addressed_temp_file<P: AsRef<Path>>(
     }
 }
 
+/// Symlinks are always available on UNIX, but on Windows they may not be
+/// enabled
+#[cfg(unix)]
+pub fn check_symlink_support() -> Result<bool, std::io::Error> {
+    Ok(true)
+}
+
+/// Symlinks are always available on UNIX, but on Windows they may not be
+/// enabled
+#[cfg(windows)]
+pub fn check_symlink_support() -> Result<bool, std::io::Error> {
+    let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+    let sideloading =
+        hklm.open_subkey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\AppModelUnlock")?;
+    let developer_mode: u32 = sideloading.get_value("AllowDevelopmentWithoutDevLicense")?;
+    Ok(developer_mode == 1)
+}
+
+#[cfg(windows)]
+pub fn try_symlink<P: AsRef<Path>, Q: AsRef<Path>>(
+    original: P,
+    link: Q,
+) -> Result<(), SymlinkError> {
+    const ERROR_PRIVILEGE_NOT_HELD: i32 = 1314;
+
+    // this will create a nonfunctional link for directories, but at the moment
+    // we don't have enough information in the tree to determine whether the
+    // symlink target is a file or a directory
+    symlink_file(original, link).map_err(|err| {
+        if err.raw_os_error() == Some(ERROR_PRIVILEGE_NOT_HELD) {
+            SymlinkError::DeveloperModeNotEnabled
+        } else {
+            SymlinkError::Io(err)
+        }
+    })
+}
+
+#[cfg(unix)]
+pub fn try_symlink<P: AsRef<Path>, Q: AsRef<Path>>(
+    original: P,
+    link: Q,
+) -> Result<(), SymlinkError> {
+    symlink(original, link).map_err(SymlinkError::Io)
+}
+
```

Comment:
nit: since there are multiple platform-specific functions, they can be organized as inner modules (if you prefer):

```rust
pub use self::platform::*;

#[cfg(unix)]
mod platform {
}

#[cfg(windows)]
mod platform {
}
```

Reply:
sure, i've done that and moved the platform-specific use statements inside the inner modules